lib/helpers.ex

```diff
@@ -117,7 +117,8 @@ defmodule EVM.Helpers do
     end
   end
 
-  @spec encode_val(integer() | list(integer())) :: list(EVM.val)
+  @spec encode_val(integer() | binary() | list(integer())) :: list(EVM.val)
+  def encode_val(n) when is_binary(n), do: :binary.decode_unsigned(n)
```

Comment:
`|> encode_val`?

Reply:
I thought so too! But we actually don't want to encode_val binary values. SHA3 for example can create binaries that "look" like negative values(ie 0xFFF...). We don't want to modify those values or it'd mess up the hash value. If we have a binary value that does actually need to represent a negative number we can encode it inline.